Data/OAuthClient/Sources/AppAuthOAuthClient.swift

```diff
@@ -0,0 +1,123 @@
+//
```

Comment:
It would be *really really* great to have some tests to validate the success cases and a couple of failure cases as well, please. This would not only help ensure the code works but also ensure it is testable as we add additional logic in the future, such as the state machine for automatically logging in a user at startup.

Reply:
Definitely this will be handled next inshaa Allah.  I usually go test-first, but for this PR the full responsibilities of the auth client weren't clear yet. As things stand, `AppAuthOAuthClient` is just a wrapper for the library. 

We have the following: 
- Token refresh & persistance
- Signaling API availability 
- Authenticating the API requests
- Signaling if OAuth is available (disables/removes the login option) 

This structure is going to be the subject of the next PR, and with it, I'll resume the standard testing strategy.

Reply:
As for the comment on managing the configuration along with the availability of the OAuth client, I'll take that into account while I handle the responsibilities mentioned above. 

The Profile service will be facing the features. 

The OAuth client will be facing other data clients, which I expect them to be used by the domain services (bookmarks, notes ...). 
The question here is where to delegate the responsibility of knowing of the API availability. I think out understanding of this will need to evolve as we handle the cases of data syncing.